### ISSUE TITLE
What is the difference between metric and eval_metric?

### ISSUE THREAD

**Issue by wangxk15** [microsoft/LightGBM#6337]
As I know, the objective function when initializing the model will be used to do the gradient descent for the model, the eval_metric when calling the fit funtion will indicate how the model will be early stopped if it didn't decrease in n continuous rounds. But what's the purpose of the metric parameter when initializing the model? Or are they actually the same and we can pass it as a parameter in either LGBMRegressor() or fit()? If so, what if we pass 2 difference eval metric in LGBMRegressor() and fit(), will on override another? I can't see any specific explanation in the documentation. 

If I want to write a custom objective function and a custom metric function so that the training will be stopped when it doesn't improve a few rounds, how should I pass my metric function to the parameter?
